onnxruntime/core/providers/openvino/openvino_execution_provider.h

```diff
@@ -13,6 +13,27 @@
 
 namespace onnxruntime {
 
+static std::vector<std::string> split(const std::string &s, char delim) {
+    std::vector<std::string> result;
+    std::stringstream ss(s);
+    std::string item;
+
+    while (getline(ss, item, delim)) {
+        result.push_back(item);
+    }
+    return result;
+}
+
+static std::vector<std::string> parseDevices(const std::string& device_string) {
```

Comment:
parseDevices should also detect invalid device names listed and accordingly return the error (maybe as an empty list). Though this is performed in the build.py, device_type may also be passed in dynamically in runtime, which won't go through the python check. We would need a C++ check here too like you have implemented in [build.py line 257](https://github.com/intel/onnxruntime/pull/47/files#diff-a545fb992b8a8781df05a2c40541f429bfff7e0d8ecc9ef0f5d559994e187814R257).

Reply:
@smkarlap  Implemented, changes pushed.